basic-auth/main.go

```diff
@@ -0,0 +1,148 @@
+package main
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os"
+	"os/signal"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/gptscript-ai/go-gptscript"
+	"github.com/tidwall/gjson"
+)
+
+type input struct {
+	ToolDisplayName string `json:"tool_display_name,omitempty"`
+	Message         string `json:"message"`
+	UsernameField   string `json:"username_field,omitempty"`
+	UsernameEnv     string `json:"username_env,omitempty"`
+	PasswordField   string `json:"password_field,omitempty"`
+	PasswordEnv     string `json:"password_env,omitempty"`
+	Metadata        map[string]string
+}
+
+type sysPromptInput struct {
+	Message   string            `json:"message,omitempty"`
+	Fields    string            `json:"fields,omitempty"`
+	Sensitive string            `json:"sensitive,omitempty"`
+	Metadata  map[string]string `json:"metadata,omitempty"`
+}
+
+func main() {
+	// Set up signal handler
+	ctx, cancel := signal.NotifyContext(context.Background(), os.Interrupt, os.Kill)
+	defer cancel()
+
+	in, err := getInput()
+	if err != nil {
+		fmt.Println("Input error: ", err)
+		os.Exit(1)
+	}
+
+	username, password, err := getCredentials(ctx, in)
+	if err != nil {
+		fmt.Println("Error getting credentials:", err)
+		os.Exit(1)
+	}
+	fmt.Printf(`{"env": {"%s": "%s", "%s": "%s"}}`, in.UsernameEnv, username, in.PasswordEnv, password)
+}
+
+func getCredentials(ctx context.Context, in input) (string, string, error) {
+	client, err := gptscript.NewGPTScript()
+	if err != nil {
+		fmt.Println("Error creating GPTScript client:", err)
+		return "", "", fmt.Errorf("Error creating GPTScript client: %w", err)
+	}
+	defer client.Close()
+
+	sysPromptIn, err := json.Marshal(sysPromptInput{
+		Message:   in.Message,
+		Fields:    strings.Join([]string{in.UsernameField, in.PasswordField}, ","),
+		Sensitive: strconv.FormatBool(true),
+		Metadata:  in.Metadata,
+	})
+	if err != nil {
+		return "", "", fmt.Errorf("Error marshalling sys prompt input: %w", err)
+	}
+
+	run, err := client.Run(ctx, "sys.prompt", gptscript.Options{
+		Input: string(sysPromptIn),
+	})
+	if err != nil {
+		return "", "", fmt.Errorf("Error running GPTScript prompt: %w", err)
+	}
+
+	res, err := run.Text()
+	if err != nil {
+		return "", "", fmt.Errorf("Error getting GPTScript response: %w", err)
+	}
+
+	username := gjson.Get(res, in.UsernameField).String()
```

Comment:
first time seeing this gjson library? does it actually throw an error if res is not valid json?